src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http3RequestStream.cs

```diff
@@ -1028,6 +1028,11 @@ private int ReadResponseContent(HttpResponseMessage response, Span<byte> buffer)
                         totalBytesRead += bytesRead;
                         _responseDataPayloadRemaining -= bytesRead;
                         buffer = buffer.Slice(bytesRead);
+
```

Comment:
This is not right. If there are any trailing headers, you'd just leave them in the stream and not read them ever.

BTW, I've investigated this area just yesterday and put a fix for something else in https://github.com/dotnet/runtime/pull/56867.